modules/serverless/pages/features/transport-encryption/transport-encryption-setup.adoc

```diff
@@ -0,0 +1,226 @@
+
+* Install the OpenShift CLI (`oc`).
+
+* Install the {serverlessoperatorname}.
+
+* Install the {certmanageroperatorname}.
+
+== Setup a `SelfSigned` `ClusterIssuer`
+
+> `ClusterIssuers`, are Kubernetes resources that represent certificate authorities (CAs) that are able to generate signed certificates by honoring certificate signing requests.
+All cert-manager certificates require a referenced issuer that is in a ready condition to attempt to honor the request. +
+Reference: https://cert-manager.io/docs/concepts/issuer/
+
+[IMPORTANT]
+====
+For the simplicity of this guide, we will use a `SelfSigned` issuer as root certificate, however, be aware of the implications and limitations as documented at https://cert-manager.io/docs/configuration/selfsigned/ of this method.
+====
+
+. Create a `SelfSigned` `ClusterIssuer`:
++
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: knative-eventing-selfsigned-issuer
+spec:
+  selfSigned: {}
+----
++
+. Apply the `ClusterIssuer` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a root certificate using the previously created `SelfSigned` `ClusterIssuer`:
++
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: knative-eventing-selfsigned-ca
+  namespace: cert-manager <1>
+spec:
+  secretName: knative-eventing-ca <2>
+
+  isCA: true
+  commonName: selfsigned-ca
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+
+  issuerRef:
+    name: knative-eventing-selfsigned-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+
+----
++
+<1> The {certmanageroperatorname} namespace, `cert-manager` by default.
+<2> Secret name that will be later used for the `ClusterIssuer` for Eventing
++
+. Apply the `Certificate` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Setup `ClusterIssuer` for Eventing
+
+. Create the `knative-eventing-ca-issuer` `ClusterIssuer` for Eventing:
++
+[source,yaml]
+----
+# This is the issuer that every Eventing component should use to issue their server's certs.
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: knative-eventing-ca-issuer
+spec:
+  ca:
+    secretName: knative-eventing-ca <1>
+----
++
+<1> Secret name in the {certmanageroperatorname} namespace (`cert-manager` by default) containing the certificate that can then be used by Knative Eventing components for new certificates.
++
+[IMPORTANT]
+====
+The name of the `ClusterIssuer` must be `knative-eventing-ca-issuer`.
+====
++
+. Apply the `ClusterIssuer` resource:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+== Understanding the transport encryption configuration
+
+The transport-encryption feature flag is an enum configuration that configures how Addressables (Broker, Channel, Sink) should or must accept events.
+
+The possible values for `transport-encryption` are:
+
+* *disabled* (this is equivalent to the current behavior)
```

Comment:
```suggestion
* *`disabled`* (this is equivalent to the current behavior)
```
(only not sure if this is supported in adoc)

Reply:
I wanted to suggest the same. I think is supported, as above we have `transport-encryption`